oqsprov/oqsprov_keys.c

```diff
@@ -403,27 +423,44 @@ EVP_PKEY *setECParams(EVP_PKEY *eck, int nid) {
                                          0x02, 0x08, 0x01, 0x01, 0x0b};
 
     const unsigned char *params;
+    size_t params_len;
+    EVP_PKEY *result = NULL;
+
+    if (eck == NULL) {
+        return NULL;
+    }
+
     switch (nid) {
     case NID_X9_62_prime256v1:
         params = p256params;
-        return d2i_KeyParams(EVP_PKEY_EC, &eck, &params, sizeof(p256params));
+        params_len = sizeof(p256params);
```

Comment:
This seems like a semantics change: Can you please explain the rationale, @songlingatpan ?